sim-cli/src/main.rs

```diff
@@ -40,33 +40,23 @@ async fn main() -> anyhow::Result<()> {
     let mut clients: HashMap<PublicKey, Arc<Mutex<dyn LightningNode + Send>>> = HashMap::new();
 
     for connection in nodes {
-        // TODO: We should simplify this into two minimal branches plus shared logging and inserting into the list
-        match connection {
-            NodeConnection::LND(c) => {
-                let node_id = c.id;
-                let lnd = LndNode::new(c).await?;
+        // TODO: Feels like there should be a better way of doing this without having to Arc<Mutex<T>>> it at this time.
+        // Box sort of works, but we won't know the size of the dyn LightningNode at compile time so the compiler will
+        // scream at us when trying to create the Arc<Mutex>> later on while adding the node to the clients map
+        let node: Arc<Mutex<dyn LightningNode + Send>> = match connection {
+            NodeConnection::LND(c) => Arc::new(Mutex::new(LndNode::new(c).await?)),
+            NodeConnection::CLN(c) => Arc::new(Mutex::new(ClnNode::new(c).await?)),
+        };
 
-                log::info!(
-                    "Connected to {} - Node ID: {}.",
-                    lnd.get_info().alias,
-                    lnd.get_info().pubkey
-                );
+        let node_info = node.lock().await.get_info().clone();
```

Comment:
Yeah, I think it's not that bad, I just feel a higher-ranked wizard may find a nicer way to do so. Looks like I'll have to go back to Hogwarts, or leave it as is until someone more talented nitpicks about it

Reply:
I'd avoid wizardry by waiting to create the mutex when inserting to clients map?

Reply:
That won't work since that's where the actual wizardry comes from.

The issue lies in that the size of `dyn LightningNode + Send` cannot be known at compilation time, and that is required to build an `Arc<Mutex<T>>`

Reply:
you're right